### PR TITLE
Restructuring the code

### PR DESCRIPTION
**Pull request by btweinstein**
Restructuring and rewriting with dimensionless units in python, cython, and opencl. There are some problems with your code @matheuscfernandes, though....I propose that if we cannot figure out the issues, we just do not use the periodic velocity inlet. Perhaps you can help me figure them out though!

### CODE REVIEW

File: LB_D2Q9/dimensionless/cython_dim.pyx [hunk 1]
```diff
@@ -0,0 +1,596 @@
...
+#         cdef float[:,:,:] farr = self.f
+#         cdef float[:] rho_w = self.rho[0, 1:ly]
+#         cdef float[:] rho_e = self.rho[lx, 1:ly]
+#
+#         cdef float[:,:,:] f = self.f
+#         cdef int ii
+#
+#         for ii in range(1,ly):
+#             # INLET: imposed velocity of u_w in the x direction and 0 in the y direction
+#
```
Comment (btweinstein) [inline LB_D2Q9/dimensionless/cython_dim.pyx:484]: Something crazy was happening here with rhos, where you were setting rho at the inlet and rho at the outlet in the move_bcs method. This is _extremely_ dangerous. You should do this in another class. I deleted the line and tried to fix it, but saw crazy things.

File: LB_D2Q9/dimensionless/cython_dim.pyx [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+#             farr[1, 0, ii] = farr[3,0,ii] + (2./3.)*rho_w[ii]*u_w
+#             farr[5, 0, ii] = farr[7,0,ii] - (1./2.)*(farr[2,0,ii]-farr[4,0,ii]) + (1./6.)*rho_w[ii]*u_w
+#             farr[8, 0, ii] = farr[6,0,ii] + (1./2.)*(farr[2,0,ii]-farr[4,0,ii]) + (1./6.)*rho_w[ii]*u_w
+#
+#             # OUTLET: imposed velocity of u_w in the x direction and 0 in the y direction
+#             rho_e[ii] = (1./(1.+u_e))*(farr[0,lx,ii]+farr[2,lx,ii]+farr[4,lx,ii]+2.*(farr[1,lx,ii]+farr[5,lx,ii]+farr[8,lx,ii]))
```
Comment (btweinstein) [inline LB_D2Q9/dimensionless/cython_dim.pyx:490]: Again, in the move_bc class, there is no way you should be updating rho's..._terrible_ things will happen.

File: LB_D2Q9/dimensionless/cython_dim.pyx [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+#
+#         u_w=self.lb_inlet_velocity
+#         u_e=self.lb_inlet_velocity
+#
+#         # INLET: define the density and prescribe the velocity
+#         u[0, 1:ly] = u_w
+#         rho[0, 1:ly] = (1./(1.-u_w))*(f[0,0,1:ly]+f[2,0,1:ly]+f[4,0,1:ly]+2.*(f[3,0,1:ly]+f[6,0,1:ly]+f[7,0,1:ly]))
+#         # OUTLET: define the density and prescribe the velocity
+#         u[lx, 1:ly] =u_e
+#         rho[lx, 1:ly] = (1./(1.+u_e))*(f[0,lx,1:ly]+f[2,lx,1:ly]+f[4,lx,1:ly]+2.*(f[1,lx,1:ly]+f[5,lx,1:ly]+f[8,lx,1:ly]))
```
Comment (btweinstein) [inline LB_D2Q9/dimensionless/cython_dim.pyx:592]: Were you trying to do this in the other method too?

File: LB_D2Q9/dimensionless/cython_dim.pyx (hunk 2 above)
Comment (matheuscfernandes) [inline LB_D2Q9/dimensionless/cython_dim.pyx:490]: I am not updating the rho's here. I am just creating an 'equivalent' rho to compute the velocity bc and what the f's have to be. I am not altering the rho's of the simulation....

**Pull request merged**